### PR TITLE
Add more details and alternative designs to the subresource explainer.

### PR DESCRIPTION
**Pull request by jyasskin**
* What makes a bundled subresource authoritative?
* How else could we express scope?
* Is the link type named right?

@hayatoito

### CODE REVIEW

File: explainers/subresource-loading.md [hunk 1]
```diff
@@ -29,22 +48,58 @@ a format that allows multiple resources to be bundled, e.g.
 - Related issues: [#411](https://github.com/WICG/webpackage/issues/411),
   [#526](https://github.com/WICG/webpackage/issues/526)
 
-## Proposal
+## Requirements
+
+Web pages will declare that some of their their subresources are provided by the
```
Comment (kinu) [inline explainers/subresource-loading.md:53]:
```suggestion
Web pages will declare that some of their subresources are provided by the
```

File: explainers/subresource-loading.md [hunk 2]
```diff
@@ -29,22 +48,58 @@ a format that allows multiple resources to be bundled, e.g.
...
+
+It's likely that the HTML parser will encounter some of the bundle's
+subresources before it receives the bundle's index. The declaration needs to
+somehow prevent the parser from double-fetching those bytes, which it can
+accomplish in a couple ways.
+
+We don't see an initial need for an associated Javascript API to pull
+information out of the bundle.
+
+We also don't address a way for Service Workers to use bundles to fill a Cache.
```
[the remark this answers is not in the document]
Comment (jyasskin) [reply explainers/subresource-loading.md:66]: I elaborated here. It's mostly that it's not necessary for a minimum-viable feature, but we can also learn more about the right API design by letting some sites unpack the bundle themselves.

File: explainers/subresource-loading.md [hunk 3]
```diff
@@ -100,13 +155,114 @@ subresources, `a.js`, `b.js`, and `c.png`, from the bundle.
...
+The approach above of just defining a scope directory doesn't allow much
+flexiblity in how resources are grouped into bundles. Several other scoping
+mechanisms are available to give the bundler more flexibility, at the cost of
+being bigger and less readable.
+
+#### Compressed list of resources
+
+As discussed in [Dynamic bundle serving with
+WebBundles](https://docs.google.com/document/d/11t4Ix2bvF1_ZCV9HKfafGfWu82zbOD7aUhZ_FyDAgmA/edit),
+simply including a base64-encoded compressed list of resources may cost as
```
[the remark this answers is not in the document]
Comment (jyasskin) [reply explainers/subresource-loading.md]: Good point. Leaving the URLs uncompressed may even help compress them against their other use in the same document.

File: explainers/subresource-loading.md [hunk 4]
```diff
@@ -100,13 +155,114 @@ subresources, `a.js`, `b.js`, and `c.png`, from the bundle.
...
+little as 5 bytes per URL on average. (That document doesn't show the details of
+that measurement.) The compressed list of resources could look like:
```
[the remark this answers is not in the document]
Comment (jyasskin) [reply explainers/subresource-loading.md]: And I've linked it here. Thanks!

File: explainers/subresource-loading.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (yoavweiss) [reply explainers/subresource-loading.md:66]: OK. I'm not 100% convinced that SW should not be part of an MVP, but agree that polyfilling an API for them may worrk as a stop gap and can help inform what such an API should look like.

**Pull request merged**